src/core/presence.ts

```diff
@@ -360,4 +346,11 @@ export class DefaultPresence implements Presence {
       return options;
     };
   }
+
+  private _assertChannelState(): void {
+    if (this._channel.state !== 'attaching' && this._channel.state !== 'attached') {
+      this._logger.error('could not perform presence operation; channel is not attached');
+      throw new Ably.ErrorInfo('could not perform presence operation; channel is not attached', 40000, 400);
```

Comment:
Channel is definitely correct but in chat should we say room not attached?

Reply:
Yeah, good shout - 3de90dc